File: src/main/java/dev/dovydasvenckus/scrapper/client/WebScrapperClient.java
package dev.dovydasvenckus.scrapper.client;

import dev.dovydasvenckus.scrapper.client.model.ScrapeRequest;
import dev.dovydasvenckus.scrapper.client.model.ScrapeResult;
import dev.dovydasvenckus.scrapper.client.service.ScrapingService;

import java.net.http.HttpClient;

public class WebScrapperClient {
    private final ScrapingService scrapingService;
    private final HttpClient client = HttpClient.newBuilder().build();

    public WebScrapperClient(String webScraperApiUrl) {
        this.scrapingService = new ScrapingService(webScraperApiUrl);
    }

    public ScrapeResult scrape(ScrapeRequest request) {
        return scrapingService.scrape(request, client);
    }
}

File: src/main/kotlin/dev/dovydasvenckus/scrapper/client/WebScrapperClient.kt
package dev.dovydasvenckus.scrapper.client

import javax.ws.rs.client.Client
import javax.ws.rs.client.ClientBuilder
import dev.dovydasvenckus.scrapper.client.jackson.KotlinObjectMapperProvider
import dev.dovydasvenckus.scrapper.client.model.ScrapeRequest
import dev.dovydasvenckus.scrapper.client.model.ScrapeResult
import dev.dovydasvenckus.scrapper.client.service.ScrapingService

class WebScrapperClient(webScraperApiUrl: String) {
    private val scrapingService: ScrapingService = ScrapingService(webScraperApiUrl)
    private val client: Client = ClientBuilder.newBuilder()
        .register(KotlinObjectMapperProvider::class.java)
        .build()

    fun scrape(request: ScrapeRequest): ScrapeResult {
        return scrapingService.scrape(request, client)
    }
}